aiida_diff/parsers.py

```diff
@@ -42,29 +40,19 @@ def parse(self, **kwargs):
         """
         from aiida.orm import SinglefileData
 
-        # Check that the retrieved folder is there
-        try:
```

Comment:
I guess it is fine to remove this, since currently the `retrieved` folder will *always* be created, even if there no files specified to be retrieved. It will just be empty. Note that this then becomes more or less unchangeable in `aiida-core`. I tried to think if there might be a use-case for ever not wanting to retrieve anything, just like it is not compulsory to run a parser, but I could not come up with anything sensible

Reply:
> I guess it is fine to remove this, since currently the retrieved folder will always be created, even if there no files specified to be retrieved. It will just be empty.

Ok!

> Note that this then becomes more or less unchangeable in aiida-core

First, I think this is independent of whether we suggest to do a try/except here or not (I guess you agree, just making sure).

Second, I agree that the vast majority of cases will require a retrieved folder, so for the moment it's OK to always create one.
However, I personally don't see a problem with, at some point, making the creation of a retrieved folder dependent on some aspects of the output spec.
People would have to do this explicitly (and adapt their parsers accordingly).

Reply:
> First, I think this statement is independent of whether we suggest to do a try/except here or not (I guess you agree, just making sure).

Well not quite. Because, if at some point we decide to not always attach the retrieved folder (or there might even be a bug in `aiida-core`) then the parser will except causing the `CalcJob` process to become `Excepted`. An exit code is always the nicer option, which is why I put the try-except there. But yes in reality it is maybe an overly cautious safety